acados/ocp_nlp/ocp_nlp_common.c

```diff
@@ -1834,6 +1841,10 @@ void ocp_nlp_approximate_qp_matrices(ocp_nlp_config *config, ocp_nlp_dims *dims,
         // init Hessian to 0 
         blasfeo_dgese(nu[i] + nx[i], nu[i] + nx[i], 0.0, mem->qp_in->RSQrq+i, 0, 0);
 
+        // Levenberg Marquadt term
+        if (opts->levenberg_marquadt > 0.0)
```

Comment:
why not moving the check into the setter and throw and error if the value is negative?

Reply:
moreover, I guess it would make most sense to scale the contribution by the discretization step length.

Reply:
> why not moving the check into the setter and throw and error if the value is negative?

There is such a check in the Python setter.
I put the if here, just to skip the operation if `levenberg_marquardt` is not used.
Not sure if it is a good idea, since it is such a cheap operation though.